### ISSUE TITLE
Add API to subscribe settled invoices

### ISSUE THREAD

**Issue by yzernik** [L2-Technology/sensei#61]
Something like the equivalent of `SubscribeInvoices` in LND: https://api.lightning.community/#subscribeinvoices

This will probably require gRPC, because it has a streaming response.

**Comment by johncantrell97** [comment]
yeah we definitely need to add support for some grpc stream apis but for each one we add it needs to also have a websocket equivalent.

**Comment by johncantrell97** [comment]
similar to how there is the generic NodeRequest/NodeResponse and AdminRequest/AdminResponse enums we will probably want generic mpsc::Channel's for sending/receiving this data over.  

I think there probably needs to be an issue about setting up streaming apis in general that needs to happen in order to properly implement this and other endpoints.

**Comment by yzernik** [comment]
An alternative way to implement this is to do what c-lightning does:
https://lightning.readthedocs.io/lightning-waitanyinvoice.7.html

> The waitanyinvoice RPC command waits until an invoice is paid, then returns a single entry ...
